stix_shifter/stix_translation/src/json_to_stix/observable.py

```diff
@@ -4,7 +4,7 @@
     'ipv6': ('(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::(ffff(:0{1,4}){0,1}:){0,1}((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9]))'),
     'mac': ('^(([0-9a-fA-F]{2}[:-]){5}([0-9a-fA-F]{2})|([0-9a-fA-F]{3}[\.]){3}([0-9a-fA-F]{3}))$'),
     'ipv4_cidr': ('^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\/[0-2][0-5]?$'),  # noqa: E501
-    'domain_name': ('^([a-z0-9]+(-[a-z0-9]+)*)*[\.a-z]*$')
+    'domain_name': ('^([a-zA-Z0-9]+(-[a-zA-Z0-9]+)*)*[\.a-zA-Z]*$|^([a-zA-Z0-9]+(-[a-zA-Z0-9]+)*\.)+[a-zA-Z]{2,}$')
```

Comment:
It looks like the old regex was allowing for domain names without the top level domain. You'll want to confirm this, but I think you could just change it to the following: `('^([a-zA-Z0-9]+(-[a-zA-Z0-9]+)*){1}(\.[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*)*(\.[a-zA-Z]{2,})+$')`

Reply:
We faced issue with AWS EC2 instance DNS name in the format ip-xxx-xx-xx-xx.ec2.internal with old REGEX. Hence we added OR conditioned REGEX. The above given RE is also working. so we will take this and update in the file.

Reply:
On running pytest, one of the carbon black test (test_merge_results_mixed_to_stix) is failing with suggested RE. So we are retaining the above OR conditioned RE itself:  'domain_name': ('^([a-zA-Z0-9]+(-[a-zA-Z0-9]+)*)*[\.a-zA-Z]*$|^([a-zA-Z0-9]+(-[a-zA-Z0-9]+)*\.)+[a-zA-Z]{2,}$')

Reply:
The CB test that's failing is on the following in the test results payload: `"hostname": "redlab-vuln2"`. That doesn't look like a valid domain name. If the hostname is getting written to the STIX domain-name object then that may mean there is a defect in the CB connector.

Reply:
Actually, the regex can be updated in a different issue. Doing a little reading, it looks like we need to account for new generic top level domains that may contain numbers and dashes. (http://newgtlds.icann.org/en/program-status/delegated-strings)

Reply:
We are adding both REGEX (old and newly suggested one) in OR (|) condition now  ('^([a-z0-9]+(-[a-z0-9]+)*)*[\.a-z]*$')|('^([a-zA-Z0-9]+(-[a-zA-Z0-9]+)*){1}(\.[a-zA-Z0-9]+(-[a-zA-Z0-9]+)*)*(\.[a-zA-Z]{2,})+$'). Hope carbonblack issue will be tracked separately. Once it is fixed, we will remove old RE ('^([a-z0-9]+(-[a-z0-9]+)*)*[\.a-z]*$') from the condition and keep one.